Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -3,7 +3,155 @@ import mongoc
+        default:
+            throw UserError.logicError(message: "target is missing")
+        }
+    }
+
+    func toTopologyTypeArray(topologies: [String]) -> [TopologyDescription.TopologyType] {
+        var topologyTypeArr: [TopologyDescription.TopologyType] = []
+        for topology in topologies {
+            topologyTypeArr.append(TopologyDescription.TopologyType(from: topology))
+        }
+        return topologyTypeArr
+    }
+
+    func testChangeStreamSpec() throws {
+        let testFilesPath = MongoSwiftTestCase.specsPath + "/change-streams/tests"
+        let testFiles: [String] = try FileManager.default.contentsOfDirectory(atPath: testFilesPath)
+
+        let tests: [ChangeStreamTestFile] = try testFiles.map { fileName in
+            let url = URL(fileURLWithPath: "\(testFilesPath)/\(fileName)")
+            var testFile = try BSONDecoder().decode(ChangeStreamTestFile.self, from: Document(fromJSONFile: url))
+
+            testFile.name = fileName
+            return testFile
+        }
+
+        let globalClient = try MongoClient(MongoSwiftTestCase.connStr)
+
+        for testFile in tests {
+            var seenError: Error?
+            let client = try MongoClient(MongoSwiftTestCase.connStr)
+            let version = try globalClient.serverVersion()
+            let topology = MongoSwiftTestCase.topologyType
+            let db1 = globalClient.db(testFile.database_name)
+            let db2 = globalClient.db(testFile.database2_name)
+            print("\n------------\nExecuting tests from file \(testFilesPath)/\(testFile.name)...\n")
+            for test in testFile.tests {
+                try db1.drop()
+                try db2.drop()
+                try db1.createCollection(testFile.collection_name)
+                try db2.createCollection(testFile.collection2_name)
+
+                let testTopologies = toTopologyTypeArray(topologies: test.topology)
+
+                guard testTopologies.contains(topology) else {
+                    print("Skipping test case because of unsupported topology type \(topology)")
+                    continue
+                }
+
+                guard version >= test.minServerVersion else {
+                    print("Skipping tests from file, minimum required server version not met.")
+                    continue
+                }
+                print("Executing test: \(test.description)")
+
+                var changeStream: ChangeStream<ChangeStreamTestEventDocument>?
+                do {
+                    changeStream = try createChangeStream(client: client, testFile: testFile, test: test)
+                } catch {
+                    seenError = error
+                }
+
+                if let failPoint = test.failPoint {
+                    try self.activateFailPoint(failPoint)
+                }
+
+                try test.run(client: globalClient, seenError: seenError, changeStream: changeStream)
+
+                // TODO: add APM assertions
```

Comment:
this TODO can be moved into the body of `run`. Also, can you file a ticket for the follow on work and add it to the TODO?